go/stellar/acctbundle/acctbundle.go

```diff
@@ -0,0 +1,673 @@
+package acctbundle
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/protocol/stellar1"
+	"github.com/stellar/go/keypair"
+	"golang.org/x/crypto/nacl/secretbox"
+)
+
+// New creates a BundleRestricted from an existing secret key.
+func New(secret stellar1.SecretKey, name string) (*stellar1.BundleRestricted, error) {
+	secretKey, accountID, _, err := libkb.ParseStellarSecretKey(string(secret))
+	if err != nil {
+		return nil, err
+	}
+	return &stellar1.BundleRestricted{
+		Revision: 1,
+		Accounts: []stellar1.BundleEntryRestricted{
+			newEntry(accountID, name, false),
+		},
+		AccountBundles: map[stellar1.AccountID]stellar1.AccountBundle{
+			accountID: newAccountBundle(accountID, secretKey),
+		},
+	}, nil
+}
+
+// NewInitial creates a BundleRestricted with a new random secret key.
+func NewInitial(name string) (*stellar1.BundleRestricted, error) {
+	full, err := keypair.Random()
+	if err != nil {
+		return nil, err
+	}
+
+	x, err := New(stellar1.SecretKey(full.Seed()), name)
+	if err != nil {
+		return nil, err
+	}
+
+	x.Accounts[0].IsPrimary = true
+
+	return x, nil
+}
+
+// NewFromBundle creates a BundleRestricted from a Bundle.
+func NewFromBundle(bundle stellar1.Bundle) (*stellar1.BundleRestricted, error) {
+	r := &stellar1.BundleRestricted{
+		Revision:       bundle.Revision,
+		Prev:           bundle.Prev,
+		AccountBundles: make(map[stellar1.AccountID]stellar1.AccountBundle),
```

Comment:
want ownhash for posting a good prev with the upgrade?

Reply:
I added it.